src/VecSim/algorithms/brute_force.cpp

```diff
@@ -247,7 +263,7 @@ extern "C" VecSimQueryResult *BruteForce_TopKQuery(VecSimIndex *index, const voi
         std::fill_n(scores, bfIndex->vectorBlockSize, 1.0);
         bfIndex->distanceCalculationFunction(dim, vectorBlock, queryBlob, scores);
         for (int i = 0; i < MIN(vectorBlock->size, k); i++) {
-            size_t min_index = cblas_ismin(vectorBlock->size, scores, 1);
+            size_t min_index = BruteForce_IndexOfMin(scores, vectorBlock->size);
```

Comment:
Can we run this outside of the loop?

Reply:
I'm trying to come up with a better approach. For now, we can't do it outside the loop as we need top-k